api.py

```diff
@@ -79,6 +83,31 @@ def post(self, url, data=None, json=None, **kwargs):
 
         return super().post(url, data=data, json=json, **kwargs)
 
+    def delete(self, url):
+        super().delete(url)
+
+        # Wait for snapshots to be deleted
+        if 'volume-snapshots' in url:
+            timeout = datetime.now() + timedelta(seconds=60)
```

Comment:
Nit: Prefer to do this with `time.monotonic`, which works almost the same. It's not that relevant here, as I doubt we run these tests during daylight savings changes, but I think it's a good habit and the code is even slightly shorter:
```python
timeout = time.monotonic() + 60

while time.monotonic() < timeout:
    ....
```